### PR TITLE
Azure Active Directory Single Sign On

### PR DESCRIPTION
**Pull request by tombuildsstuff**
- Adds support for logins using Azure Active Directory

Shout-out to @Nepomuceno @at10ti0n and @andyroyle for their work on this

### CODE REVIEW

File: client/configuration/config.js [hunk 1]
```diff
@@ -1,5 +1,5 @@
...
+.constant('ENV', {RequiresAuth:true,etcdHost:'127.0.0.1',etcdPort:'4001',hobknobHost:'localhost',hobknobPort:'3006',AuthProviders:{AzureAuth:{AzureClientId:'',AzureClientSecret:'',AzureTenantId:''}},categories:[{id:0,name:'Simple Features',description:'Use when you want your feature to be either on or off'},{id:1,name:'Domain Features',description:'Use when you want your features to be toggled separately for different domains (e.g. com, couk, fr, ...)',values:['com','couk','de','fr']},{id:2,name:'Locale Features',description:'Use when you want your features to be toggled separately for different locales (e.g. en-GB, en-US, fr-FR, ...)',values:['en-GB','en-US','fr-FR','de-DE']}],plugin:{path:'../server/examplePlugin.js'}})
```
Comment (tombuildsstuff) [inline client/configuration/config.js]: this needs reverting to 'false'

File: config/config.json [hunk 2]
```diff
@@ -1,9 +1,18 @@
 {
-    "RequiresAuth": false,
+    "RequiresAuth": true,
```
Comment (tombuildsstuff) [inline config/config.json]: this needs reverting to 'false'

File: config/config.json [hunk 3]
```diff
@@ -1,9 +1,18 @@
...
     "hobknobPort": "3006",
+    "AuthProviders":
+        {
+            "AzureAuth":
+            {
+                "AzureClientId": "",
+                "AzureClientSecret":"",
+                "AzureTenantId": ""
+            }
+        },
```
Comment (tombuildsstuff) [inline config/config.json]: we should probably add the relevant google support if that's required/optional here?

Comment (andyroyle) [inline config/config.json]: the default config can just omit the `AuthProviders` property

**Pull request merged**